src/app/app.component.ts

```diff
@@ -12,14 +12,18 @@ export class AppComponent {
     alert("I did stuff");
   }
 
-  doOtherStuff() {
-    try {
-      for (let i = 0; i < 50; i++) {
-        console.log('looping for no good reason');
-      }
-      alert("I did other stuff");
-    } catch (e) {
-      throw e;
+  public doOtherStuff() {
+    var a = [1, 2, 3];
+    var b = a.reverse(); // Noncompliant
+  }
+
+  public foo(a: boolean, b: boolean) {
+    if (a) {
+      console.log('a!');
+    } if (b) {
+      console.log('b!');
     }
+    let i = 0;
+    i = i++;
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this increment or correct the code not to waste it. ([typescript:S2123](http://localhost:9076/coding_rules#rule_key=typescript%3AS2123))

[See it in SonarQube](http://localhost:9076/project/issues?id=org.sonarqube%3Aangular-sample&pullRequest=1&issues=AWmQHKGAZQKnx6JGnxzk&open=AWmQHKGAZQKnx6JGnxzk)

[ISSUE_KEY]: AWmQHKGAZQKnx6JGnxzk